### PR TITLE
kci_test update_rootfs_urls

### PR DESCRIPTION
**Pull request by gctucker**
Add a command to automatically generate the YAML configuration with the rootfs images URLs when they exist and leave them to their previous URLs when getting HTTP 404.  For example:
```
kci_test update_rootfs_url --release=20220913.0 --output=config/core/rootfs-image.yaml
```

### CODE REVIEW

File: kci_test [hunk 1]
```diff
@@ -73,42 +77,114 @@ class cmd_validate(Command):
...
+        yield fs, config, name, root_type
+
+
+def _check_rootfs_urls(fs_config, rootfs_config, root_type, verbose):
+    url_formats = list(
+        url_fmt for url_fmt in (
+            fs_config.get_url_format(fmt_type)
+            for fmt_type in set({root_type}).union({'nfs', 'ramdisk'})
+        ) if url_fmt
+    )
```
Comment (mgalka) [inline kci_test:100]: I think we can just use a normal list comprehension here

```suggestion
    url_formats = [
        url_fmt for url_fmt in (
            fs_config.get_url_format(fmt_type)
            for fmt_type in set({root_type}).union({'nfs', 'ramdisk'})
        ) if url_fmt
    ]
```

The internal one can be also converted to list comprehension or you can leave it as a generator expression as it is now. Having said that there's not much gain from using genexp there.

**Review by mgalka — COMMENTED** [review]
one small comment

File: kci_test (hunk 1 above)
Comment (gctucker) [reply kci_test:100]: Yes I think that's some leftover from previous experiments, I'll change it to `[]`

**Comment by gctucker** [comment]
@aliceinwire FYI this is moving all the CIP test configuration to a separate `test-configs-cip.yaml` file as the CIP rootfs config can't be automatically generated like the regular buildroot and debos ones which are built by KernelCI.  It might cause some trivial conflicts with other PRs, I hope that's OK.

**Review by mgalka — APPROVED** [review]

**Pull request merged**